tests/foreman/cli/test_host.py

```diff
@@ -2076,6 +2078,188 @@ def test_negative_multi_cv_registration(
+    assert rhel_contenthost.subscribed
+
+    # Unregister the host
+    unregister_result = rhel_contenthost.unregister()
+    assert unregister_result.status == 0
+
+    # Set the allow_multiple_content_views setting to true
+    session_multicv_sat.cli.Settings.set({'name': 'allow_multiple_content_views', 'value': 'true'})
+
+    # Register the host with subscription-manager, passing multiple environments
+    env_names = f"{library_lce.name}/{cv1.name},{multicv_lce.name}/{cv2.name}"
+    res = rhel_contenthost.register_contenthost(multicv_org.label, lce=None, environments=env_names)
+
+    # Confirm that registration succeeds
+    assert res.status == 0
+    assert rhel_contenthost.subscribed
+
+    # Confirm that the host is registered to both environments
+    host = session_multicv_sat.cli.Host.info({'name': rhel_contenthost.hostname})
+    assert (
+        len(host['content-information']['content-view-environments']) == 2
+    ), "Expected host to be registered to both environments"
+
+
+@pytest.mark.rhel_ver_match('[^7]')
+def test_positive_multi_cv_host_repo_availability(
+    session_multicv_sat,
+    rhel_contenthost,
+):
+    """Multi-environment hosts should have access to repositories from all of their content view environments.
+
+    :id: 6a15d591-be84-4b5b-8deb-8ea6eb32fee6
+
+    :steps:
+        1. Create two synced custom repositories
+        2. Add each repository to its own content view. Publish and promote the content views.
+        3. Register a host with global registration to a single content view environment
+        4. Expose the allow_multiple_content_views setting
+        5. Set the allow_multiple_content_views setting to true
+        6. Assign the host to multiple content view environments
+        7. Confirm repos listed in subscription-manager repos match the repos in the content view environments
+
+    :expectedresults: Host sees repositories from both content view environments, not just one.
+
+    :CaseImportance: Critical
+
+    :parametrized: yes
+    """
+
+    # Create an org
+    multicv_org = session_multicv_sat.api.Organization().create()
+
+    # Create activation key
+    multicv_ak = session_multicv_sat.api.ActivationKey(
+        content_view=multicv_org.default_content_view.id,
+        environment=multicv_org.library.id,
+        organization=multicv_org,
+    ).create()
+
+    # Create a lifecycle environment
+    multicv_lce = session_multicv_sat.api.LifecycleEnvironment(organization=multicv_org).create()
+
+    # Create a content view
+    cv1 = session_multicv_sat.api.ContentView(organization=multicv_org).create()
+    cv1.publish()
+    cv1 = cv1.read()
+
+    # Create two synced custom repositories
+    repo_instances = []
+    for repo in ['RepoA', 'RepoB']:
+        product_name = f'{repo}_Product_{str(uuid.uuid4())}'
+        repo_id = session_multicv_sat.api_factory.create_sync_custom_repo(
+            org_id=multicv_org.id,
+            product_name=product_name,
```

Comment:
```suggestion
            product_name=gen_string('alpha'),
```
I would just go with `gen_string` if we don't need to use/know the name later and if there is no need to have this particular name form

Reply:
I wanted to avoid this since those random alphanumeric names are so unreadable. This is one of my dislikes of Robottelo. I wonder if some day Robottelo can switch to a random-word-based name generator instead of random letters.

Reply:
These names make it harder to debug. For example, if the test is expecting "RepoA_ProductA" and getting "RepoA_ProductB" it's a lot easier to see that than it is to see "sxijfediwhero4gti9hyuuj0w9u3t" vs "wi8het30o3t895y723jhsdwgj".

Reply:
I agree this is pain we suffer while debugging tests. On the other hand there would be more than one tests using RepoA and ProductA for sure. (I know, isolation per random-named org should be sufficient in this case)